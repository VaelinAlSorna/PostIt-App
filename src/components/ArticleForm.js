import React from 'react';
import moment from 'moment';

class ArticleForm extends React.Component {
    state = {
        title: '',
        text: '',
        createdAt: moment().format('D. M. Y')
    };
    onTitleChange = (e) => {
        const title = e.target.value;
        this.setState(() => ({
            title
        }))
    }
    onTextChange = (e) => {
        const text = e.target.value;
        this.setState(() => ({
            text
        }))
    }
    onSubmit = (e) => {
        e.preventDefault();
        this.props.onSubmit(this.state)
    }
    render () {
       return ( 
        <div> 
            <form onSubmit={this.onSubmit}> 
                <input 
                    onChange={this.onTitleChange} 
                    placeholder="Title" 
                    type="text" 
                    value={this.state.title}
                />
                <textarea 
                    onChange={this.onTextChange}
                    placeholder="Add text for your article"
                    value={this.state.text}
                >
                </textarea>
                <input type="submit" value="Submit" />
            </form>
        </div>
       );
    }
}

export default ArticleForm;