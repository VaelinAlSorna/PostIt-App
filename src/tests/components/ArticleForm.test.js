import React from 'react';
import { shallow } from 'enzyme';
import ArticleForm from '../../components/ArticleForm';
import articles from '../fixtures/articlesFixtures';

test('should render ArticleForm correctly with default data', () => {
    const wrapper = shallow(<ArticleForm />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ArticleForm correctly with article data', () => {
    const wrapper = shallow(<ArticleForm article={articles[0]} />);
    expect(wrapper).toMatchSnapshot();
}); 

test('should call onSubmit when correct data is provided', () => {
    const onSubmitSpy = jest.fn()
    const wrapper = shallow(<ArticleForm article={articles[0]} onSubmit={onSubmitSpy}/>);
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
        title: articles[0].title,
        text: articles[0].text,
        createdAt: articles[0].createdAt
    });
});

test('should print error when when  form is submitted with incorrect data', () => {
    const wrapper = shallow(<ArticleForm />);
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(wrapper.state('error')).toBe('Your title and text must contain at least 3 characters');
});

test('should change title on input onChange', () => {
    const value = 'Testing '
    const wrapper = shallow(<ArticleForm />);
    wrapper.find('input').at(0).simulate('change',{
        target: { value }
    });
    expect(wrapper.state('title')).toBe(value);
});

test('should change text on textarea onChange', () => {
    const value = 'Some text for testing';
    const wrapper = shallow(<ArticleForm />);
    wrapper.find('textarea').simulate('change',{
        target: { value }
    });
    expect(wrapper.state('text')).toBe(value);
});