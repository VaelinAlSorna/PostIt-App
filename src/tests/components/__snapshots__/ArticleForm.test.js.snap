// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ArticleForm correctly with article data 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Title"
    type="text"
    value="Breaking news"
  />
  <textarea
    onChange={[Function]}
    placeholder="Add text for your article"
    value="Political info"
  />
  <input
    type="submit"
    value="Save Article"
  />
</form>
`;

exports[`should render ArticleForm correctly with default data 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Title"
    type="text"
    value=""
  />
  <textarea
    onChange={[Function]}
    placeholder="Add text for your article"
    value=""
  />
  <input
    type="submit"
    value="Save Article"
  />
</form>
`;
